llama.cpp

```diff
@@ -2147,14 +2147,15 @@ struct llama_control_vector {
 };
 
 struct llama_vocab {
-    using id    = int32_t;
-    using token = std::string;
-    using ttype = llama_token_type;
+    using id       = int32_t;
+    using token    = std::string;
+    using ttype    = llama_token_type;
+    using tattribs = llama_token_attribs;
```

Comment:
minor : might want to rename like so:

```suggestion
    using tattr = llama_token_attr;
```

But feel free to ignore this suggestion